### DISCUSSION TITLE
Feature request - automatic download

### DISCUSSION THREAD

**Discussion by Bahazzie** [nzbgetcom/nzbget#253] [Ideas]
Hello people,

As a movie lover, currently what we miss is automatic downloads. Let me explain.

- As we all now, 99% of posted movies is totally unwatchable
- So I spent years with spotnet, manually checking all new posts with IMDB, and downloading only the good ones
- It's SOOO repetitive, therefore I automated this: wrote a java program that fetches latest 100 movies posts from website nzbserver.com. Then it checks each movie with IMDB and if rating > threshold the program clicks the download link on the NZBSERVER site that places NZB in NZBGET folder. It also checks for max sixe, not too old, and excluded genres. And of course already downloaded. The thresholds (min IMDB, min Year, ExcludedGenres are in a small config file)

This has actually run succesfully for years now. the java program runs every 10 minutes on a small old NUC box and downloads only the NZB's  of what has at least a CHANCE of being watchable. Except for one problem: NZBServer.com generated NZB's do not contain encryption password, they somehow wipe it out of the metadata. So still some manual post processing for me to do and I am getting tired of it.

For years I have wondered why this functionality is not available. Could it be an idea to include in the revived NZBGet? I give you the concrete specs below:

- Run program every PARAMETER minutes to check for new posts for movies
- Check each new post if IMDB rating is > PARAMETER
- If rating can not be found set default rating = PARAMETER  (set 0 for never download unrated, 9 for always)
- Check MaxSize is < PARAMETER
- Check MinYear > PARAMETER
- Check genre not in PARAMETER (example: KOREAN, MUSICAL, WESTERN)
- Write motivated decisions to logfile if PARAMETER debug is set

In my experience, even by setting IMDB limit > 6 there is still a lot of junk coming in but a factor 100 less :-) Isn't this a world class idea? Programs as Sonar/Radar only look out for given titles. I want everything NEW that has a chance of being good. 

Otherwise, if this is can not be in scope, what how can I fetch new posts and NZB's that DO contain the passwordsin the metadata that NZBSERVER seems to filter out? Then I do it myself, beit in JAVA.

Bas

**Comment by Bahazzie** [comment]
Hey! I see a solution for my problem...where nzbserver.com wipes pw out of the NZB the alternative nzbclub (same content)  does not!

**Comment by luckedea** [comment]
NZBGet can run binary extensions! :)

I can relate. For me it's a perl script that does similar, but also including shows. I didn't ditch a simple manual moderation step for items it finds. I haven't seen a comparable tool, except maybe a good recommendations engine?

**Reply by Bahazzie** [reply]
Well at least my problem is solved it seems...nzbserver.com strips passwords from the generated nzb, but clubnzb.com happily leaves them in for the same content. As for recommendations, yeah in my  case it generates recommendations (first calls webservice OMDB for rating) and then automatically downloads all of them every 10 minutes :-) Still junk comes through, that I manually delete. But I stopped fetching unrated movies else you drown in d-movies.

**Comment by Bahazzie** [comment]
Here is a fragment from my log maybe for inspiration :-)

{main] INFO org..imdb.Scraper - processing movie: De Bende Van Vier (2008)
[main] INFO org..imdb.Scraper - Failed criterieum excludedGenres with Cabaret
[main] INFO org..imdb.Scraper - processing movie: Vermines aka Infested (2023)
[main] INFO org..imdb.Scraper - Failed criterieum imdbscore Limit with 4.0
[main] INFO org..imdb.Scraper - processing movie: Dusk For A Hitman 2023 1080p AMZN WEB-DL DDP5 1 H 264-GP-M-NLsubs
[main] INFO org..imdb.Scraper - Failed criterieum already downloaded
[main] INFO org..imdb.Scraper - processing movie: Billy Turf
[main] INFO org..imdb.Scraper - Failed criterium yearLimit with 1900
[main] INFO org..imdb.Scraper - processing movie: Django 1966 2160p UHD BluRay H265-MALUS
[main] INFO org..imdb.Scraper - Failed criterium yearLimit with 1966
[main] INFO org..imdb.Scraper - processing movie: Blood For Dust 2023 1080p AMZN WEB-DL DDP5 1 H 264-GP-M-NLsubs
[main] INFO org..imdb.Scraper - Failed criterieum already downloaded
[main] INFO org..imdb.Scraper - processing movie: Honeymoonish (2024) 1080p NF WEB-DL DD+5.1 Retail NLsub
[main] INFO org..imdb.Scraper - Failed criterieum imdbscore Limit with 4.0
[main] INFO org..imdb.Scraper - processing movie: Peter Pan & Wendy (2023) 1080p DSNP WEB-DL DUAL DD+5.1 Atmos NL 
[main] INFO org..imdb.Scraper - Failed criterieum imdbscore Limit with 4.4
[main] INFO org..imdb.Scraper - processing movie: The Fall Guy 2024 1080p Clean Cam X264 COLLECTIVE
[main] INFO org..imdb.Scraper - Failed criterieum already downloaded
[main] INFO org..imdb.Scraper - processing movie: Curious Caterer Foiled Plans 2024 1080p WEB-DL DDP5 1 H 264-GP-M-Eng
[main] INFO org..imdb.Scraper - Failed criterieum imdbscore Limit with 4.0
[main] INFO org..imdb.Scraper - written to processed.txt
[main] INFO org..imdb.Scraper - going to sleep for 600 seconds
